modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java

```diff
@@ -0,0 +1,934 @@
+    private static final String PRIMARY_IS_SP_CALLBACK_URL =
+            "http://localhost:8490/playground2/oauth2-authorize-user.jsp";
+    private static final String PRIMARY_IS_SP_BACK_CHANNEL_LOGOUT_URL = "http://localhost:8490/playground2/bclogout";
+
+    // Urls related to service provider configured in Federated IS.
+    private static final String FEDERATED_IS_SP_NAME = "travelocity";
+    private static final String FEDERATED_IS_SP_CALLBACK_URL = "http://localhost:8490/travelocity.com/home.jsp";
+    private static final String FEDERATED_IS_SP_BACK_CHANNEL_LOGOUT_URL =
+            "http://localhost:8490/travelocity.com/bclogout";
+
+    private static final String OIDC_APP_CLIENT_ID = "ClientID";
+    private static final String OIDC_APP_CLIENT_SECRET = "ClientSecret";
+
+    // Username and password of the user in federated IS.
+    private static final String FEDERATED_IS_TEST_USERNAME = "testFederatedUser";
+    private static final String FEDERATED_IS_TEST_PASSWORD = "testFederatePassword";
+    private static final String FEDERATED_IS_TEST_USER_ROLES = "admin";
+
+    private Map<Integer, ApplicationManagementServiceClient> applicationManagementServiceClients;
+    private Map<Integer, OauthAdminClient> oAuthAdminClients;
+    private IdentityProviderMgtServiceClient identityProviderMgtServiceClient;
+    private TenantManagementServiceClient tenantServiceClient;
+    private UserManagementClient usrMgtClient;
+    private ServerConfigurationManager serverConfigurationManager;
+    private final AutomationContext context;
+    // Client Id and Secret of primary Is service provider configured in federated idp.
+    private String fedISClientID;
+    private String fedISClientSecret;
+    // Client Id and Secret of service provider configured in primary Is.
+    private String primSPClientID;
+    private String primSPClientSecret;
+    // Client Id and Secret of service provider configured in federated Is.
+    private String fedSP_ClientID;
+    private String fedSP_ClientSecret;
+    // Data related to the login flow.
+    private String username;
+    private String primaryIsk;
+    private String federatedIsk;
+    private String primaryUserId;
+    private String federatedUserId;
+    private CookieStore cookieStore;
+    private CloseableHttpClient client;
+    private HttpClient httpClientWithoutAutoRedirections;
+    private JSONParser jsonParser;
+
+    @DataProvider(name = "configProvider")
+    public static Object[][] configProvider() {
+
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN}};
+    }
+
+    @Factory(dataProvider = "configProvider")
+    public OIDCFederatedIdpInitLogoutTest(TestUserMode userMode) throws Exception {
+
+        context = new AutomationContext("IDENTITY", userMode);
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initTest() throws Exception {
+
+        super.init();
+//        changeISConfiguration();
```

Comment:
why it is commented? We need to start the server with tenant qualified URL enabled